### ISSUE TITLE
html.security.audit.missing-integrity.missing-integrity false positive when <link> element has rel="preconnect"

### ISSUE THREAD

**Issue by creste** [returntocorp/sgrep-rules#2835]
**Describe the bug**
This code:
```html
<link rel="preconnect" href="https://fonts.gstatic.com/" />
```
is flagged by the `html.security.audit.missing-integrity.missing-integrity` rule.  It is a false positive because `rel="preconnect"` is a hint to browsers that the user is likely to need resources from the target resource's origin.  No actual content is downloaded, thus adding an [integrity](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity) string doesn't make sense and shouldn't be flagged by the `html.security.audit.missing-integrity.missing-integrity` rule.

**To Reproduce**
Run the [missing-integrity](https://github.com/returntocorp/semgrep-rules/blob/9f363237880de1ac965b32c8546f3e68f1447686/html/security/audit/missing-integrity.yaml) rule against the HTML code shown above.

**Expected behavior**
No issues are found.

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [X] P2: annoying but not blocking me

**Additional context**
I think these patterns are the root cause:
https://github.com/returntocorp/semgrep-rules/blob/9f363237880de1ac965b32c8546f3e68f1447686/html/security/audit/missing-integrity.yaml#L28-L33

**Comment by inkz** [comment]
Thanks @creste for reporting this. Fix created at https://github.com/returntocorp/semgrep-rules/pull/2842
